### ISSUE TITLE
[BUG] Auto-scrolling when not at the bottom

### ISSUE THREAD

**Issue by alxkum** [aws-samples/amazon-bedrock-client-for-mac#57]
**Describe the bug**
Usually apps will not keep auto-scrolling when the user has scrolled up. When text output is still being streamed and you want to start reading, you scroll up, but the app will keep scrolling down. So you have to wait until the full output is generated to even start reading.

**To Reproduce**
1. Create a prompt that results in somewhat larger output, e.g., `list top 100 music artists of all time`.
2. Try reading from the beginning while the output is still being generated. You won't be able to.

**Expected behavior**
The "auto-scrolling" behavior should only occur when the vertical scrollbar is at the bottom already (or doesn't exist yet because the content fits into the viewport).

**Environment (required):**
 - macOS version: 13.7.1
 - Amazon Bedrock Client for Mac version: 1.2.4

**Comment by sbhal** [comment]
Yup, this is annoying for long outputs, essentially making streaming response useless while you wait for full response.

**Comment by didhd** [comment]
This issue has been addressed in commit 9498218 and released in v1.2.6 (12f6194).

The changes include:
- Added anchor-based scroll detection to prevent auto-scrolling when users manually scroll up
- Maintained auto-scroll functionality only when at the bottom of the chat
- Added layout delay to improve smoothness and prevent flickering

This should resolve the original issue by allowing users to read previous content while new messages are being streamed.

Please update to v1.2.6 or later to get these improvements.

/cc @alxkum @sbhal

**Comment by alxkum** [comment]
Awesome, thank you!  

One tiny thing though: it seems like the inner scrollbar gets offset by half whenever you scroll up now and that new arrow-down overlay button appears:  
![Screenshot 2024-12-23 at 11 38 25](https://github.com/user-attachments/assets/21b75b0a-6376-4049-837e-828e57f71e8b)  

Here's what it looks like when at the bottom:  
![Screenshot 2024-12-23 at 11 39 01](https://github.com/user-attachments/assets/ca21857f-dcd8-4da2-a76b-2992bfcd5303)

I'm an macOS 13.7.2

**Comment by didhd** [comment]
reopening the issue

**Comment by didhd** [comment]
The scrollbar thickness issue identified by @alxkum has been fully resolved in our [release v1.2.7](https://github.com/aws-samples/amazon-bedrock-client-for-mac/releases/tag/v1.2.7).

This was specifically addressed in commit [39a45c4](https://github.com/aws-samples/amazon-bedrock-client-for-mac/commit/39a45c4b3ccf469f82f123afc173d6bce6fd286c) "fix(ui): resolve scroll thickness issue caused by arrow-down button overlay" and further improved in the comprehensive UI/UX overhaul.

The fix ensures that:
- The scrollbar maintains consistent thickness throughout the UI
- The arrow-down button overlay no longer affects scrollbar positioning
- No horizontal scrollbar appears unnecessarily

Thank you for your patience and for reporting this issue with detailed screenshots. The improvements to the chat scroll behavior now provide a much better experience when reading content while responses are still streaming.

Closing this issue as fixed in v1.3.0.
